src/circuit/gadget/ecc/chip/double.rs

```diff
@@ -0,0 +1,103 @@
+use super::{util, CellValue, EccConfig, EccPoint};
+
+use halo2::{
+    arithmetic::FieldExt,
+    circuit::Region,
+    plonk::{ConstraintSystem, Error, Expression},
+};
+
+pub(super) fn create_gate<F: FieldExt>(
+    meta: &mut ConstraintSystem<F>,
+    q_double: Expression<F>,
+    x_a: Expression<F>,
+    y_a: Expression<F>,
+    x_p: Expression<F>,
+    y_p: Expression<F>,
+) {
+    let x_p_2 = x_p.clone() * x_p.clone();
+    let x_p_4 = x_p_2.clone() * x_p_2.clone();
+
+    // 4⋅(y_p)^2⋅(x_a + 2⋅x_p) − 9⋅(x_p)^4 = 0
```

Comment:
Where does this equation come from?

Reply:
It comes from substituting `lambda` in the point doubling formulae. I've added a commit documenting this in the ECC book PR: https://github.com/zcash/orchard/pull/54/commits/58e3da88c274fa0261d61d550dfec224e74311b4

Reply:
Resolving this conversation since `double.rs` has been removed.